OpenRA.Mods.Common/Activities/PickupUnit.cs

```diff
@@ -129,6 +118,25 @@ public override bool Tick(Actor self)
 			return TickChild(self) && state == PickupState.Pickup;
 		}
 
+		public override void Cancel(Actor self, bool keepQueue = false)
+		{
+			base.Cancel(self, keepQueue);
+
+			if (!IsInterruptible)
+				return;
+
+			if (carryall.State == Carryall.CarryallState.Reserved)
+				carryall.UnreserveCarryable(self);
```

Comment:
This is the exact reason I didn't move this in https://github.com/OpenRA/OpenRA/pull/20489, before we unreserved on cargo death in all cases and not just when the state is reserved.

Reply:
Looks like it was added to fix https://github.com/OpenRA/OpenRA/pull/16509#issuecomment-496034776 (I can't link a deleted accounts message).

After investigation to me it seems like a redundant check, shall we just remove it?

Reply:
Removed. I played around even more with it and found no reason for this check to exist

Reply:
Wouldn't that mean that we unreserve the carryable even when we're carrying it?

Reply:
No, because the activity is not interuptable on takeoff

Reply:
Where is this set? The original takeoff activity only has priority.

Reply:
I had more debugging done, although it appeared to be working correct on surface, it was cancelling slightly incorrectly. Fixed